### PR TITLE
Feat/ncurses lib

### LINKED ISSUE #35: Implement NCurses lib
# :white_check_mark: What's needed
Create NCURSESlib class to create the .so

## 📋 Subtasks
- [x] NCURSESlib class

### PR DESCRIPTION
**Pull request by theophile-jr**
Add ncurses lib basics

### CODE REVIEW

File: src/Graphical/NCURSES/ncurses.cpp [hunk 1]
```diff
@@ -5,19 +5,66 @@
...
+
+void NCURSESlib::clearWindow() {
+    clear();
+};
+
+void NCURSESlib::displayWindow() {
+    refresh();
+};
+
+int NCURSESlib::getKeyEvent() {
```
Comment (mathieurobert1) [inline src/Graphical/NCURSES/ncurses.cpp:40]: Where is the call of the _Keymap ?

**Review by mathieurobert1 — COMMENTED** [review]
There just the _Keymap which is not use, is it normal ?

File: src/Graphical/NCURSES/getInstance.cpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+#include <iostream>
+#include "ncurses.hpp"
+
+extern "C"
+{
+    __attribute__((constructor))
+    static void initsharedlibrary()
+    {
+        std::cout << "Loading foo ncurses ..." << std::endl;
```
Comment (Thomaltarix) [inline src/Graphical/NCURSES/getInstance.cpp]: You should change Foo to ncurses i think

Comment (Thomaltarix) [reply src/Graphical/NCURSES/getInstance.cpp]: (Like Loading ncurses library ...)

**Review by Thomaltarix — COMMENTED** [review]
I agree with the comments of Mathieu, but i'll accept it as soon as its fixed

**Review by Thomaltarix — APPROVED** [review]
Clean

**Review by mathieurobert1 — APPROVED** [review]
Seems good :+1:

**Pull request merged**